File: frontend/src/types/aftersale.ts
/**
 * 售后相关类型定义
 * 定义售后管理模块的数据类型和接口
 */

// 故障报告
export interface FaultReport {
  faultID: string      // 故障ID
  partID: string       // 关联零部件
  vin: string          // 车辆VIN码
  reporter: string     // 上报者ID
  description: string  // 故障描述
  faultType: string    // AI分类结果
  riskProb: number     // 风险概率
  reportTime: string   // 上报时间
  status: string       // 处理状态
}

// 召回记录
export interface RecallRecord {
  recallID: string       // 召回ID
  batchNos: string[]     // 涉及批次列表
  reason: string         // 召回原因
  affectedParts: string[] // 受影响零部件ID列表
  status: string         // 进度
  createTime: string     // 召回发起时间
  finishTime: string     // 完成时间
}

// 售后记录
export interface AftersaleRecord {
  aftersaleID: string      // 售后记录唯一ID
  partID: string           // 关联零部件
  batchNo: string          // 关联批次
  vin: string              // 关联车辆
  faultID: string          // 关联故障上报
  recallID: string         // 关联召回记录
  aftersaleType: string    // 售后类型
  aftersaleStatus: string  // 处理状态
  handlerOrgID: string     // 处理机构ID
  handlerID: string        // 处理人ID
  description: string      // 售后描述
  processResult: string    // 处理结果
  processTime: string      // 处理完成时间
  cost: string             // 售后成本
  remark: string           // 售后备注
}

// 创建故障报告请求
export interface CreateFaultReportRequest {
  faultID: string
  partID: string
  vin: string
  reporter: string
  description: string
  faultType?: string
  riskProb?: number
  status?: string
}

// 创建召回记录请求
export interface CreateRecallRecordRequest {
  recallID: string
  batchNos: string[]
  reason: string
  affectedParts: string[]
  status?: string
}

// 创建售后记录请求
export interface CreateAftersaleRecordRequest {
  aftersaleID: string
  partID: string
  batchNo: string
  vin: string
  faultID?: string
  recallID?: string
  aftersaleType: string
  aftersaleStatus?: string
  handlerOrgID: string
  handlerID: string
  description: string
  processResult?: string
  cost?: string
  remark?: string
}

File: backend/model/aftersale.go
package model

import "encoding/json"

type FaultReport struct {
	FaultID     string  `json:"faultID"`
	PartID      string  `json:"partID"`
	VIN         string  `json:"vin"`
	Reporter    string  `json:"reporter"`
	Description string  `json:"description"`
	FaultType   string  `json:"faultType"`
	RiskProb    float64 `json:"riskProb"`
	ReportTime  string  `json:"reportTime"`
	Status      string  `json:"status"`
}

type FaultReportDTO struct {
	FaultID     string  `json:"faultID" binding:"required"`
	PartID      string  `json:"partID" binding:"required"`
	VIN         string  `json:"vin"`
	Reporter    string  `json:"reporter" binding:"required"`
	Description string  `json:"description" binding:"required"`
	FaultType   string  `json:"faultType"`
	RiskProb    float64 `json:"riskProb"`
	Status      string  `json:"status"`
}

type RecallRecord struct {
	RecallID      string   `json:"recallID"`
	BatchNos      []string `json:"batchNos"`
	Reason        string   `json:"reason"`
	AffectedParts []string `json:"affectedParts"`
	Status        string   `json:"status"`
	CreateTime    string   `json:"createTime"`
	FinishTime    string   `json:"finishTime"`
}

// UnmarshalJSON 自定义反序列化，处理 affectedParts 为 null 的情况
func (r *RecallRecord) UnmarshalJSON(data []byte) error {
	type Alias RecallRecord
	aux := &struct {
		AffectedParts []string `json:"affectedParts"`
		*Alias
	}{
		Alias: (*Alias)(r),
	}
	if err := json.Unmarshal(data, &aux); err != nil {
		return err
	}
	if aux.AffectedParts == nil {
		r.AffectedParts = []string{}
	} else {
		r.AffectedParts = aux.AffectedParts
	}
	return nil
}

type RecallRecordDTO struct {
	RecallID      string   `json:"recallID" binding:"required"`
	BatchNos      []string `json:"batchNos" binding:"required"`
	Reason        string   `json:"reason" binding:"required"`
	AffectedParts []string `json:"affectedParts"`
	Status        string   `json:"status"`
}

type AftersaleRecord struct {
	AftersaleID     string      `json:"aftersaleID"`
	PartID          string      `json:"partID"`
	BatchNo         string      `json:"batchNo"`
	VIN             string      `json:"vin"`
	FaultID         string      `json:"faultID"`
	RecallID        string      `json:"recallID"`
	AftersaleType   string      `json:"aftersaleType"`
	AftersaleStatus string      `json:"aftersaleStatus"`
	HandlerOrgID    string      `json:"handlerOrgID"`
	HandlerID       string      `json:"handlerID"`
	Description     string      `json:"description"`
	ProcessResult   string      `json:"processResult"`
	ProcessTime     string      `json:"processTime"`
	Cost            string      `json:"cost,omitempty"`
	AttachmentInfo  *Attachment `json:"attachmentInfo,omitempty"`
	Remark          string      `json:"remark,omitempty"`
}

type AftersaleRecordDTO struct {
	AftersaleID     string      `json:"aftersaleID" binding:"required"`
	PartID          string      `json:"partID" binding:"required"`
	BatchNo         string      `json:"batchNo"`
	VIN             string      `json:"vin"`
	FaultID         string      `json:"faultID"`
	RecallID        string      `json:"recallID"`
	AftersaleType   string      `json:"aftersaleType" binding:"required"`
	AftersaleStatus string      `json:"aftersaleStatus"`
	HandlerOrgID    string      `json:"handlerOrgID"`
	HandlerID       string      `json:"handlerID"`
	Description     string      `json:"description"`
	ProcessResult   string      `json:"processResult"`
	ProcessTime     string      `json:"processTime"`
	Cost            string      `json:"cost,omitempty"`
	AttachmentInfo  *Attachment `json:"attachmentInfo,omitempty"`
	Remark          string      `json:"remark,omitempty"`
}
